## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
| mint-fs.sh | :x: [more...](#4e12d18d-b82e-4b68-b2a7-0e528c60bfee) |
| mint-erasure.sh | :x: [more...](#cf68e1f8-07b2-4563-bcdc-211cd1d99c38) |
<a name="cf68e1f8-07b2-4563-bcdc-211cd1d99c38"></a>
### 16346-28d3c2f/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.8:31988
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp e85f5eb0c570:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-6yafe4hyfbg62o",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 271,
  "error": "RequestError: send request failed\ncaused by: Put \"http://15.15.15.8:31988/aws-sdk-go-test-6yafe4hyfbg62o\": dial tcp 15.15.15.8:31988: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-java tests ... FAILED in 2 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-6yafe4hyfbg62o",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 271,
  "error": "RequestError: send request failed\ncaused by: Put \"http://15.15.15.8:31988/aws-sdk-go-test-6yafe4hyfbg62o\": dial tcp 15.15.15.8:31988: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-php tests ... FAILED in 7 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-6yafe4hyfbg62o",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 271,
  "error": "RequestError: send request failed\ncaused by: Put \"http://15.15.15.8:31988/aws-sdk-go-test-6yafe4hyfbg62o\": dial tcp 15.15.15.8:31988: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-ruby tests ... FAILED in 6 seconds
{
  "name": "aws-sdk-ruby",
  "function": "listBuckets",
  "args": {},
  "duration": 2169.3,
  "error": "Failed to open TCP connection to 15.15.15.8:31988 (Connection refused - connect(2) for \"15.15.15.8\" port 31988)",
  "status": "FAIL"
}
(1/14) Running awscli tests ... FAILED in 14 seconds
{"name": "awscli", "duration": 12453, "function": "make_bucket\n", "status": "FAIL", "error": "Could not connect to the endpoint URL: "http://15.15.15.8:31988/awscli-mint-test-bucket-24767""}
(1/14) Running healthcheck tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": null,
  "duration": 0,
  "error": "Get \"http://15.15.15.8:31988/minio/health/live\": dial tcp 15.15.15.8:31988: connect: connection refused",
  "function": "testLivenessEndpoint",
  "message": "GET request failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(1/14) Running mc tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": null,
  "duration": 0,
  "error": "Get \"http://15.15.15.8:31988/minio/health/live\": dial tcp 15.15.15.8:31988: connect: connection refused",
  "function": "testLivenessEndpoint",
  "message": "GET request failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(1/14) Running minio-go tests ... FAILED in 7 seconds
{
  "args": {
    "bucketName": "minio-go-test-ipmptwnlmp31kbho",
    "region": "eu-west-1"
  },
  "duration": 6451,
  "error": "Put \"http://15.15.15.8:31988/minio-go-test-ipmptwnlmp31kbho/\": dial tcp 15.15.15.8:31988: connect: connection refused",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testMakeBucketErrorV2",
  "status": "FAIL"
}
(1/14) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 73,
  "status": "FAIL",
  "error": "java.net.ConnectException: Failed to connect to /15.15.15.8:31988 >>> [okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:297), okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:207), okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:226), okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:106), okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:74), okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:255), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:517), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(1/14) Running minio-js tests ... FAILED in 2 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 1012,
  "status": "FAIL",
  "error": "Error: connect ECONNREFUSED 15.15.15.8:31988 at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1159:16)"
}
(1/14) Running minio-py tests ... FAILED in 8 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-ce467c04-6d96-4806-85d4-5bc9e8a89fe1",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='15.15.15.8', port=31988): Max retries exceeded with url: /minio-py-test-ce467c04-6d96-4806-85d4-5bc9e8a89fe1 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7ff2f86ac610>: Failed to establish a new connection: [Errno 111] Connection refused'))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 174, in _new_conn\n    conn = connection.create_connection(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 95, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 85, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 703, in urlopen\n    httplib_response = self._make_request(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 398, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 239, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/usr/lib/python3.8/http/client.py\", line 1256, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1302, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1251, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1011, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.8/http/client.py\", line 951, in send\n    self.connect()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 205, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 186, in _new_conn\n    raise NewConnectionError(\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7ff2f86ac610>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 168, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 618, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 787, in urlopen\n    retries = retries.increment(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='15.15.15.8', port=31988): Max retries exceeded with url: /minio-py-test-ce467c04-6d96-4806-85d4-5bc9e8a89fe1 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7ff2f86ac610>: Failed to establish a new connection: [Errno 111] Connection refused'))\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 8047
}
(1/14) Running s3cmd tests ... FAILED in 1 seconds
{
  "name": "s3cmd",
  "duration": "268",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "ERROR: [Errno 111] Connection refused"
}
(1/14) Running s3select tests ... FAILED in 6 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-87e47758-c40f-4dfb-8acf-ea38ef538274"
  },
  "duration": 6019,
  "message": "HTTPConnectionPool(host='15.15.15.8', port=31988): Max retries exceeded with url: /s3select-test-87e47758-c40f-4dfb-8acf-ea38ef538274 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f62d18f1370>: Failed to establish a new connection: [Errno 111] Connection refused'))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 174, in _new_conn\n    conn = connection.create_connection(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 95, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 85, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 703, in urlopen\n    httplib_response = self._make_request(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 398, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 239, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/usr/lib/python3.8/http/client.py\", line 1256, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1302, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1251, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1011, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.8/http/client.py\", line 951, in send\n    self.connect()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 205, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 186, in _new_conn\n    raise NewConnectionError(\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f62d18f1370>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 95, in test_csv_input_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 618, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 787, in urlopen\n    retries = retries.increment(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='15.15.15.8', port=31988): Max retries exceeded with url: /s3select-test-87e47758-c40f-4dfb-8acf-ea38ef538274 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f62d18f1370>: Failed to establish a new connection: [Errno 111] Connection refused'))\n",
  "status": "FAIL"
}
(1/14) Running versioning tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-51gi53vek6ra54"
  },
  "duration": 1324,
  "error": "RequestError: send request failed\ncaused by: Put \"http://15.15.15.8:31988/versioning-test-51gi53vek6ra54\": dial tcp 15.15.15.8:31988: connect: connection refused",
  "function": "/mint/build/versioning/bucket.go:47: main.testMakeBucket",
  "message": "Versioning CreateBucket Failed",
  "name": "versioning",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="4e12d18d-b82e-4b68-b2a7-0e528c60bfee"></a>
### 16346-28d3c2f/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.2:30795
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp dd8d31e0ee17:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-6xpk4cnb3e2zxx",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 284,
  "error": "RequestError: send request failed\ncaused by: Put \"http://15.15.15.2:30795/aws-sdk-go-test-6xpk4cnb3e2zxx\": dial tcp 15.15.15.2:30795: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-java tests ... FAILED in 2 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-6xpk4cnb3e2zxx",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 284,
  "error": "RequestError: send request failed\ncaused by: Put \"http://15.15.15.2:30795/aws-sdk-go-test-6xpk4cnb3e2zxx\": dial tcp 15.15.15.2:30795: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-php tests ... FAILED in 6 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-6xpk4cnb3e2zxx",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 284,
  "error": "RequestError: send request failed\ncaused by: Put \"http://15.15.15.2:30795/aws-sdk-go-test-6xpk4cnb3e2zxx\": dial tcp 15.15.15.2:30795: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-ruby tests ... FAILED in 7 seconds
{
  "name": "aws-sdk-ruby",
  "function": "listBuckets",
  "args": {},
  "duration": 2163.19,
  "error": "Failed to open TCP connection to 15.15.15.2:30795 (Connection refused - connect(2) for \"15.15.15.2\" port 30795)",
  "status": "FAIL"
}
(1/14) Running awscli tests ... FAILED in 11 seconds
{"name": "awscli", "duration": 9586, "function": "make_bucket\n", "status": "FAIL", "error": "Could not connect to the endpoint URL: "http://15.15.15.2:30795/awscli-mint-test-bucket-22310""}
(1/14) Running healthcheck tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": null,
  "duration": 0,
  "error": "Get \"http://15.15.15.2:30795/minio/health/live\": dial tcp 15.15.15.2:30795: connect: connection refused",
  "function": "testLivenessEndpoint",
  "message": "GET request failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(1/14) Running mc tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": null,
  "duration": 0,
  "error": "Get \"http://15.15.15.2:30795/minio/health/live\": dial tcp 15.15.15.2:30795: connect: connection refused",
  "function": "testLivenessEndpoint",
  "message": "GET request failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(1/14) Running minio-go tests ... FAILED in 7 seconds
{
  "args": {
    "bucketName": "minio-go-test-e5pbgc3xhredl6pe",
    "region": "eu-west-1"
  },
  "duration": 7355,
  "error": "Put \"http://15.15.15.2:30795/minio-go-test-e5pbgc3xhredl6pe/\": dial tcp 15.15.15.2:30795: connect: connection refused",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testMakeBucketErrorV2",
  "status": "FAIL"
}
(1/14) Running minio-java tests ... FAILED in 2 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 68,
  "status": "FAIL",
  "error": "java.net.ConnectException: Failed to connect to /15.15.15.2:30795 >>> [okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:297), okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:207), okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:226), okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:106), okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:74), okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:255), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:517), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(1/14) Running minio-js tests ... FAILED in 1 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 1022,
  "status": "FAIL",
  "error": "Error: connect ECONNREFUSED 15.15.15.2:30795 at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1159:16)"
}
(1/14) Running minio-py tests ... FAILED in 9 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-99586291-0c99-4ff1-beea-164c7a1b3f3b",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='15.15.15.2', port=30795): Max retries exceeded with url: /minio-py-test-99586291-0c99-4ff1-beea-164c7a1b3f3b (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fb318a39610>: Failed to establish a new connection: [Errno 111] Connection refused'))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 174, in _new_conn\n    conn = connection.create_connection(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 95, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 85, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 703, in urlopen\n    httplib_response = self._make_request(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 398, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 239, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/usr/lib/python3.8/http/client.py\", line 1256, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1302, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1251, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1011, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.8/http/client.py\", line 951, in send\n    self.connect()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 205, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 186, in _new_conn\n    raise NewConnectionError(\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7fb318a39610>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 168, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 618, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 787, in urlopen\n    retries = retries.increment(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='15.15.15.2', port=30795): Max retries exceeded with url: /minio-py-test-99586291-0c99-4ff1-beea-164c7a1b3f3b (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fb318a39610>: Failed to establish a new connection: [Errno 111] Connection refused'))\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 8041
}
(1/14) Running s3cmd tests ... FAILED in 0 seconds
{
  "name": "s3cmd",
  "duration": "279",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "ERROR: [Errno 111] Connection refused"
}
(1/14) Running s3select tests ... FAILED in 7 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-e87f4777-cf32-4361-826e-5bd0455e0245"
  },
  "duration": 6019,
  "message": "HTTPConnectionPool(host='15.15.15.2', port=30795): Max retries exceeded with url: /s3select-test-e87f4777-cf32-4361-826e-5bd0455e0245 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f1c4d6153a0>: Failed to establish a new connection: [Errno 111] Connection refused'))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 174, in _new_conn\n    conn = connection.create_connection(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 95, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 85, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 703, in urlopen\n    httplib_response = self._make_request(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 398, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 239, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/usr/lib/python3.8/http/client.py\", line 1256, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1302, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1251, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1011, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.8/http/client.py\", line 951, in send\n    self.connect()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 205, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 186, in _new_conn\n    raise NewConnectionError(\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f1c4d6153a0>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 95, in test_csv_input_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 618, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 787, in urlopen\n    retries = retries.increment(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='15.15.15.2', port=30795): Max retries exceeded with url: /s3select-test-e87f4777-cf32-4361-826e-5bd0455e0245 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f1c4d6153a0>: Failed to establish a new connection: [Errno 111] Connection refused'))\n",
  "status": "FAIL"
}
(1/14) Running versioning tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-x5o1u6jtkot3p2"
  },
  "duration": 1378,
  "error": "RequestError: send request failed\ncaused by: Put \"http://15.15.15.2:30795/versioning-test-x5o1u6jtkot3p2\": dial tcp 15.15.15.2:30795: connect: connection refused",
  "function": "/mint/build/versioning/bucket.go:47: main.testMakeBucket",
  "message": "Versioning CreateBucket Failed",
  "name": "versioning",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/erasure-healing.go
+++ b/cmd/erasure-healing.go
@@ -235,6 +235,10 @@ func listAllBuckets(ctx context.Context, storageDisks []StorageAPI, healBuckets 
 				// we ignore disk not found errors
 				return nil
 			}
+			if storageDisks[index].Healing() != nil {
+				// we ignore disks under healing
+				return nil
+			}
 			volsInfo, err := storageDisks[index].ListVols(ctx)
 			if err != nil {
 				return err

--- a/cmd/erasure-server-pool.go
+++ b/cmd/erasure-server-pool.go
@@ -1721,17 +1721,7 @@ func (z *erasureServerPools) purgeDelete(ctx context.Context, bucket, prefix str
 // sort here just for simplification. As per design it is assumed
 // that all buckets are present on all serverPools.
 func (z *erasureServerPools) ListBuckets(ctx context.Context, opts BucketOptions) (buckets []BucketInfo, err error) {
-	for idx, pool := range z.serverPools {
-		if z.IsSuspended(idx) {
-			continue
-		}
-		buckets, err = pool.ListBuckets(ctx, opts)
-		if err != nil {
-			logger.LogIf(ctx, err)
-			continue
-		}
-		break
-	}
+	buckets, err = z.s3Peer.ListBuckets(ctx, opts)
 	if err != nil {
 		return nil, err
 	}

--- a/cmd/erasure-sets.go
+++ b/cmd/erasure-sets.go
@@ -26,7 +26,6 @@ import (
 	"math/rand"
 	"net/http"
 	"reflect"
-	"sort"
 	"strings"
 	"sync"
 	"time"
@@ -745,56 +744,6 @@ func (s *erasureSets) IsCompressionSupported() bool {
 
 func (s *erasureSets) IsTaggingSupported() bool {
 	return true
-}
-
-// List all buckets from one of the set, we are not doing merge
-// sort here just for simplification. As per design it is assumed
-// that all buckets are present on all sets.
-func (s *erasureSets) ListBuckets(ctx context.Context, opts BucketOptions) (buckets []BucketInfo, err error) {
-	var listBuckets []BucketInfo
-	healBuckets := map[string]VolInfo{}
-	for _, set := range s.sets {
-		// lists all unique buckets across drives.
-		if err := listAllBuckets(ctx, set.getDisks(), healBuckets, s.defaultParityCount); err != nil {
-			return nil, err
-		}
-	}
-
-	// include deleted buckets in listBuckets output
-	deletedBuckets := map[string]VolInfo{}
-
-	if opts.Deleted {
-		for _, set := range s.sets {
-			// lists all deleted buckets across drives.
-			if err := listDeletedBuckets(ctx, set.getDisks(), deletedBuckets, s.defaultParityCount); err != nil {
-				return nil, err
-			}
-		}
-	}
-	for _, v := range healBuckets {
-		bi := BucketInfo{
-			Name:    v.Name,
-			Created: v.Created,
-		}
-		if vi, ok := deletedBuckets[v.Name]; ok {
-			bi.Deleted = vi.Created
-		}
-		listBuckets = append(listBuckets, bi)
-	}
-	for _, v := range deletedBuckets {
-		if _, ok := healBuckets[v.Name]; !ok {
-			listBuckets = append(listBuckets, BucketInfo{
-				Name:    v.Name,
-				Deleted: v.Created,
-			})
-		}
-	}
-
-	sort.Slice(listBuckets, func(i, j int) bool {
-		return listBuckets[i].Name < listBuckets[j].Name
-	})
-
-	return listBuckets, nil
 }
 
 // listDeletedBuckets lists deleted buckets from all disks.

--- a/cmd/peer-s3-client.go
+++ b/cmd/peer-s3-client.go
@@ -80,6 +80,51 @@ func NewS3PeerSys(endpoints EndpointServerPools) *S3PeerSys {
 	}
 }
 
+// ListBuckets lists buckets across all servers and returns a possible consistent view
+func (sys *S3PeerSys) ListBuckets(ctx context.Context, opts BucketOptions) (buckets []BucketInfo, err error) {
+	g := errgroup.WithNErrs(len(sys.peerClients))
+
+	localBuckets, err := listBucketsLocal(ctx, opts)
+	if err != nil {
+		return nil, err
+	}
+
+	nodeBuckets := make([][]BucketInfo, len(sys.peerClients)+1)
+	errs := []error{nil}
+	nodeBuckets[0] = localBuckets
+
+	for idx, client := range sys.peerClients {
+		idx := idx
+		client := client
+		g.Go(func() error {
+			if client == nil {
+				return errPeerOffline
+			}
+			localBuckets, err := client.ListBuckets(ctx, opts)
+			if err != nil {
+				return err
+			}
+			nodeBuckets[idx+1] = localBuckets
+			return nil
+		}, idx)
+	}
+
+	errs = append(errs, g.Wait()...)
+
+	quorum := (len(sys.allPeerClients) / 2)
+	if err = reduceReadQuorumErrs(ctx, errs, bucketOpIgnoredErrs, quorum); err != nil {
+		return nil, err
+	}
+
+	for idx, buckets := range nodeBuckets {
+		if errs[idx] == nil {
+			return buckets, nil
+		}
+	}
+
+	return []BucketInfo{}, nil
+}
+
 // GetBucketInfo returns bucket stat info about bucket on disk across all peers
 func (sys *S3PeerSys) GetBucketInfo(ctx context.Context, bucket string, opts BucketOptions) (binfo BucketInfo, err error) {
 	g := errgroup.WithNErrs(len(sys.peerClients))
@@ -125,6 +170,21 @@ func (sys *S3PeerSys) GetBucketInfo(ctx context.Context, bucket string, opts Buc
 	}
 
 	return bucketInfo, nil
+}
+
+func (client *peerS3Client) ListBuckets(ctx context.Context, opts BucketOptions) ([]BucketInfo, error) {
+	v := url.Values{}
+	v.Set(peerS3BucketDeleted, strconv.FormatBool(opts.Deleted))
+
+	respBody, err := client.call(peerS3MethodListBuckets, v, nil, -1)
+	if err != nil {
+		return nil, err
+	}
+	defer xhttp.DrainBody(respBody)
+
+	var buckets []BucketInfo
+	err = gob.NewDecoder(respBody).Decode(&buckets)
+	return buckets, err
 }
 
 // GetBucketInfo returns bucket stat info from a peer

--- a/cmd/peer-s3-server.go
+++ b/cmd/peer-s3-server.go
@@ -22,6 +22,7 @@ import (
 	"encoding/gob"
 	"errors"
 	"net/http"
+	"sort"
 
 	"github.com/gorilla/mux"
 	"github.com/minio/minio/internal/logger"
@@ -41,6 +42,7 @@ const (
 	peerS3MethodMakeBucket    = "/make-bucket"
 	peerS3MethodGetBucketInfo = "/get-bucket-info"
 	peerS3MethodDeleteBucket  = "/delete-bucket"
+	peerS3MethodListBuckets   = "/list-buckets"
 )
 
 const (
@@ -75,6 +77,54 @@ func (s *peerS3Server) IsValid(w http.ResponseWriter, r *http.Request) bool {
 // HealthHandler - returns true of health
 func (s *peerS3Server) HealthHandler(w http.ResponseWriter, r *http.Request) {
 	s.IsValid(w, r)
+}
+
+func listBucketsLocal(ctx context.Context, opts BucketOptions) (buckets []BucketInfo, err error) {
+	quorum := (len(globalLocalDrives) / 2)
+
+	buckets = make([]BucketInfo, 0, 32)
+	healBuckets := map[string]VolInfo{}
+
+	// lists all unique buckets across drives.
+	if err := listAllBuckets(ctx, globalLocalDrives, healBuckets, quorum); err != nil {
+		return nil, err
+	}
+
+	// include deleted buckets in listBuckets output
+	deletedBuckets := map[string]VolInfo{}
+
+	if opts.Deleted {
+		// lists all deleted buckets across drives.
+		if err := listDeletedBuckets(ctx, globalLocalDrives, deletedBuckets, quorum); err != nil {
+			return nil, err
+		}
+	}
+
+	for _, v := range healBuckets {
+		bi := BucketInfo{
+			Name:    v.Name,
+			Created: v.Created,
+		}
+		if vi, ok := deletedBuckets[v.Name]; ok {
+			bi.Deleted = vi.Created
+		}
+		buckets = append(buckets, bi)
+	}
+
+	for _, v := range deletedBuckets {
+		if _, ok := healBuckets[v.Name]; !ok {
+			buckets = append(buckets, BucketInfo{
+				Name:    v.Name,
+				Deleted: v.Created,
+			})
+		}
+	}
+
+	sort.Slice(buckets, func(i, j int) bool {
+		return buckets[i].Name < buckets[j].Name
+	})
+
+	return buckets, nil
 }
 
 func getBucketInfoLocal(ctx context.Context, bucket string, opts BucketOptions) (BucketInfo, error) {
@@ -184,6 +234,24 @@ func makeBucketLocal(ctx context.Context, bucket string, opts MakeBucketOptions)
 	return reduceWriteQuorumErrs(ctx, errs, bucketOpIgnoredErrs, (len(globalLocalDrives)/2)+1)
 }
 
+func (s *peerS3Server) ListBucketsHandler(w http.ResponseWriter, r *http.Request) {
+	if !s.IsValid(w, r) {
+		return
+	}
+
+	bucketDeleted := r.Form.Get(peerS3BucketDeleted) == "true"
+
+	buckets, err := listBucketsLocal(r.Context(), BucketOptions{
+		Deleted: bucketDeleted,
+	})
+	if err != nil {
+		s.writeErrorResponse(w, err)
+		return
+	}
+
+	logger.LogIf(r.Context(), gob.NewEncoder(w).Encode(buckets))
+}
+
 // GetBucketInfoHandler implements peer BuckeInfo call, returns bucket create date.
 func (s *peerS3Server) GetBucketInfoHandler(w http.ResponseWriter, r *http.Request) {
 	if !s.IsValid(w, r) {
@@ -253,4 +321,5 @@ func registerPeerS3Handlers(router *mux.Router) {
 	subrouter.Methods(http.MethodPost).Path(peerS3VersionPrefix + peerS3MethodMakeBucket).HandlerFunc(httpTraceHdrs(server.MakeBucketHandler))
 	subrouter.Methods(http.MethodPost).Path(peerS3VersionPrefix + peerS3MethodDeleteBucket).HandlerFunc(httpTraceHdrs(server.DeleteBucketHandler))
 	subrouter.Methods(http.MethodPost).Path(peerS3VersionPrefix + peerS3MethodGetBucketInfo).HandlerFunc(httpTraceHdrs(server.GetBucketInfoHandler))
+	subrouter.Methods(http.MethodPost).Path(peerS3VersionPrefix + peerS3MethodListBuckets).HandlerFunc(httpTraceHdrs(server.ListBucketsHandler))
 }

